app/recordtransfer/views/transfer.py

```diff
@@ -169,6 +212,13 @@ def dispatch(self, request: HttpRequest, *args, **kwargs) -> HttpResponse:
         if not self.in_progress_submission:
             return redirect("recordtransfer:transfer")
 
+        # Check if associated upload session is expired or not
+        if (
+            self.in_progress_submission.upload_session
+            and self.in_progress_submission.upload_session.expired
+        ):
+            return redirect("recordtransfer:in_progress_submission_expired")
```

Comment:
This logical statement can be moved to the In-progress submission model. I'm also including an `upload_session_expires_soon` function here because it can be used in the template.

```python
class InProgressSubmission(models.Model):
    # ...

    def upload_session_expired(self) -> bool:
        """Determine if the associated upload session has expired or not."""
        return self.upload_session is not None and self.upload_session.expired

    def upload_session_expires_soon(self) -> bool:
        """Determine if the associated upload session is expiring soon or not."""
        return self.upload_session is not None and self.upload_session.expires_soon

    # ...
```

```suggestion
        # Check if associated upload session is expired or not
        if self.in_progress_submission.upload_session_expired():
            return redirect("recordtransfer:in_progress_submission_expired")
```